app/users/signals.py

```diff
@@ -0,0 +1,23 @@
+"""
+signals for users app
+"""
+
+from django.db.models.signals import post_delete
+from django.dispatch import receiver
+
+from app.users.models.Admin import Admin
+from app.users.models.Architect import Architect
+from app.users.models.Client import Client
+from app.users.models.Supplier import Supplier
+
+
+@receiver(post_delete, sender=Admin)
+@receiver(post_delete, sender=Architect)
+@receiver(post_delete, sender=Client)
+@receiver(post_delete, sender=Supplier)
+def delete_associated_user(sender, instance, **kwargs):
+    """
+    Deletes the associated ArchimatchUser when the user type instance is deleted.
+    """
+    if instance.user:
+        instance.user.delete()
```

Comment:
don't delete the user add a field is_deleted and test on it

Reply:
why keep the user and set is_deleted , 
if so every presence of user model should be modified to check if is_deleted , and when readding user also
why not just delete it ?

Reply:
we should keep it for data integrity and auditing ,accidental deletion